commit ee44876942df0f22d2b5db1f4caab1547c66f9cf
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2019-05-21T16:29:17+00:00 (5 months ago)
    
    TRITON-1686 linux-prepare-image support for fixing ubuntu serial console
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/tools/prepare-image/linux-prepare-image b/tools/prepare-image/linux-prepare-image
index eb3f06a..570df44 100755
--- a/tools/prepare-image/linux-prepare-image
+++ b/tools/prepare-image/linux-prepare-image
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2018, Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 #
@@ -175,6 +175,18 @@ function prepare_fedora() {
 function prepare_ubuntu() {
     # Clean up package cache
     apt-get -y clean
+
+    local fixconsole=$(mdata-get prepare-image:fix-ubuntu-console 2>/dev/null)
+
+    if [[ $fixconsole == true ]]; then
+        cat >/etc/default/grub.d/51-smartos-serial-console.cfg <<EOF
+GRUB_TERMINAL="serial console"
+GRUB_CMDLINE_LINUX="console=ttyS0,115200n8 console=tty0"
+GRUB_SERIAL_COMMAND="serial --speed=115200 --unit=0 --word=8 --parity=no --stop=1"
+EOF
+
+        update-grub2
+    fi
 }
 
 # Makes sure that /lib/smartdc et al are sane.
